src/components/BookmarksPanel.tsx

```diff
@@ -0,0 +1,116 @@
+/** ****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import { observer } from 'mobx-react-lite';
+import moment from 'moment';
+import React from 'react';
+import { getTimestampAsNumber } from '../helpers/date';
+import { isEventMessage } from '../helpers/event';
+import { createBemElement } from '../helpers/styleCreators';
+import { useSelectedStore } from '../hooks';
+import { EventAction } from '../models/EventAction';
+import { EventMessage } from '../models/EventMessage';
+
+interface SavedItem {
+	value: EventMessage | EventAction;
+	type: 'pinned-message' | 'event';
+}
+
+const BookmarksPanel = () => {
+	const selectedStore = useSelectedStore();
+	const savedItems: SavedItem[] = React.useMemo(() => {
+		return ([...selectedStore.attachedMessages, ...selectedStore.pinnedMessages] as (
+			| EventAction
```

Comment:
why do we need this type casting?

Reply:
We need this casting until we get pinned messages functionality. It shows that the array contains events and messages

Reply:
i still don't get it. either way you can create observable pinnedEvents, initialize it with empty array and add functionality later